javascript/javascript3/week2/Task4_Fetching_Waiting/fetching_waiting.js

```diff
@@ -0,0 +1,38 @@
+// TASK 4 Fetching and Waiting
+/*  The 3 steps:
+    Wait 3 seconds
+    After 3 seconds fetch some data from any api you like
+    Log out the data from the api */
+
+// USING PROMISES AND .then
+
+url = "https://api.chucknorris.io/jokes/random";
+
+function delayFetch() {
+  setTimeout(() => {
+    fetch(url)
+      .then((response) => response.json())
+      .then((data) => {
+        console.log(data.value);
+      }, 3000)
```

Comment:
There is a compile time error on line 17.
```
Argument of type '3000' is not assignable to parameter of type '((reason: any) => PromiseLike<never>) | null | undefined'.ts(2345.
```

Reply:
I have moved the delay time of 3000 to its place so now the error you are getting should be fixed.